benchmark/src/mutex.rs

```diff
@@ -57,6 +57,55 @@ impl<T> Mutex<T> for parking_lot::Mutex<T> {
     }
 }
 
+#[cfg(not(windows))]
+type SrwLock<T> = std::sync::Mutex<T>;
+
+#[cfg(windows)]
+use winapi::um::synchapi;
+#[cfg(windows)]
+struct SrwLock<T>(UnsafeCell<T>, UnsafeCell<synchapi::SRWLOCK>);
+#[cfg(windows)]
+unsafe impl<T> Sync for SrwLock<T> {}
+#[cfg(windows)]
+unsafe impl<T: Send> Send for SrwLock<T> {}
+#[cfg(windows)]
+impl<T> Mutex<T> for SrwLock<T> {
+    fn new(v: T) -> Self {
+        let mut h: synchapi::SRWLOCK = synchapi::SRWLOCK { Ptr: std::ptr::null_mut() };
+
+        unsafe {
+            synchapi::InitializeSRWLock(&mut h);
+        }
+        SrwLock(
+            UnsafeCell::new(v),
+            UnsafeCell::new(h),
+        )
+    }
+    fn lock<F, R>(&self, f: F) -> R
+    where
+        F: FnOnce(&mut T) -> R,
+    {
+        unsafe {
+            synchapi::AcquireSRWLockExclusive(self.1.get());
+            let res = f(&mut *self.0.get());
+            synchapi::ReleaseSRWLockExclusive(self.1.get());
+            res
+        }
+    }
+    fn name() -> &'static str {
+        "winapi_srwlock"
+    }
+}
+#[cfg(windows)]
+impl<T> Drop for SrwLock<T> {
+    fn drop(&mut self) {
+        // TODO: Do we need to do anything here? CloseHandle?
```

Comment:
Nothing is needed here. You can remove the TODO.

Reply:
Force pushed this change